src/hooks/useActionDelCol.js

```diff
@@ -0,0 +1,21 @@
+const useActionDelCol = (gridNodes, setGridNodes) => {
+  const handleSelect = () => { 
+    let newGridNodes = Array.from(gridNodes);
+
+    for (let i = newGridNodes.length - 1; i >= 0; i--) {
+      newGridNodes[i].pop();
```

Comment:
Samething as above. The inner array is just a reference rather than a copy. Instead, you want to create a copy of the entire row first, such as:
`newGridNodes[i] = [...newGridNodes[i]]`

And then do the pop operation